atomics/include/desul/atomics/Lock_Array_CUDA.hpp

```diff
@@ -108,27 +108,23 @@ __device__ inline void unlock_address_cuda(void* ptr, desul::MemoryScopeNode) {
   atomicExch(&desul::Impl::CUDA_SPACE_ATOMIC_LOCKS_NODE[offset], 0);
 }
 
-// Make lock_array_copied an explicit translation unit scope thingy
-namespace {
-static int lock_array_copied = 0;
-}  // namespace
-
 #ifdef __CUDACC_RDC__
 inline
 #else
 inline static
 #endif
     void
     copy_cuda_lock_arrays_to_device() {
-  if (lock_array_copied == 0) {
+  static bool lock_array_copied = []() {
```

Comment:
```suggestion
  static bool once = []() {
```

I'd prefer if you renamed in both